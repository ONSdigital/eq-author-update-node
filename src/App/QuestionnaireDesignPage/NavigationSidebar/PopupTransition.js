import { CSSTransition } from "react-transition-group";
import styled from "styled-components";
import PropTypes from "prop-types";

const PopupTransition = styled(CSSTransition).attrs({
  classNames: "popup",
})`
  opacity: 1;
  transform: translate(0, 0%);
  transition: all 0.6s -0.1s;
  transform-origin: top center;

  &.popup-enter {
    transform: translate(0, -100%);
    transition: all 0.6s 0s;
    transform-origin: top center;
  }

  &.popup-exit {
    opacity: 0.9;
    transform: translate(0, -130%);
    transition: all 0.6s 0s;
    transform-origin: top center;
  }
`;

PopupTransition.defaultProps = {
  in: false,
  timeout: 200,
};

PopupTransition.propTypes = {
  timeout: PropTypes.number,
  children: PropTypes.element,
};

export default PopupTransition;
