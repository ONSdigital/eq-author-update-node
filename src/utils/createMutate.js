const createMutate = (client, mutation) => (options) =>
  client.mutate({
    mutation,
    ...options,
  });

export default createMutate;
