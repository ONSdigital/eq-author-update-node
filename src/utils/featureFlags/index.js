// import config from "config";

export const enableOn = (flags) => {
  // const enabledFlags = config.REACT_APP_FEATURE_FLAGS.split(",");

  // const displayFeature = flags.every((flag) => enabledFlags.includes(flag));

  // return displayFeature;

  return true;
};

export const disableOn = (flags) => !enableOn(flags);
