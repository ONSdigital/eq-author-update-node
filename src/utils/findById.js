import { find } from "lodash";
const findById = (collection, id) => find(collection, { id });

export default findById;
