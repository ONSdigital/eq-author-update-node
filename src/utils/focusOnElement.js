const focusOnElement = (id) => document.getElementById(id).focus();

export default focusOnElement;
