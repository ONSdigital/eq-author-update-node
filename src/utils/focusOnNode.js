const focusOnNode = (ref) => {
  if (ref && ref.current) {
    ref.current.focus();
  }
};

export default focusOnNode;
