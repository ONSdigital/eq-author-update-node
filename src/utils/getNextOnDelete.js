import { findIndex } from "lodash";

const getNextOnDelete = (items, id) => {
  const index = findIndex(items, { id });
  let nextIndex;

  if (index === 0) {
    nextIndex = items.length > 1 ? 1 : 0;
  } else {
    nextIndex = index - 1;
  }

  return items[nextIndex];
};

export default getNextOnDelete;
