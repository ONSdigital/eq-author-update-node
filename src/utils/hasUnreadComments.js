/*
  comments: Array of comments from the page
  userId: ID of the user currently signed in (from MeContext)
*/
const hasUnreadComments = (comments, userId) => {
  let hasNotRead = false;
  if (!comments) {
    hasNotRead = false;
    return hasNotRead;
  }
  comments.forEach((comment) => {
    // If signed in user has not read the comment
    if (!comment.readBy.some((id) => userId === id)) {
      hasNotRead = true;
    }
    comment.replies.forEach((reply) => {
      // If signed in user has not read the reply
      if (!reply.readBy.some((id) => userId === id)) {
        hasNotRead = true;
      }
    });
  });
  return hasNotRead;
};

export default hasUnreadComments;
