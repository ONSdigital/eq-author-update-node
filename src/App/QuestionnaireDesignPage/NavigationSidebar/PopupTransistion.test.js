import React from "react";
import { render } from "tests/utils/rtl";

import PopupTransition from "./PopupTransition";

describe("components/PopupTransition", () => {
  it("should render", () => {
    expect(
      render(
        <PopupTransition>
          <div>Content</div>
        </PopupTransition>
      ).asFragment()
    ).toMatchSnapshot();
  });
});
