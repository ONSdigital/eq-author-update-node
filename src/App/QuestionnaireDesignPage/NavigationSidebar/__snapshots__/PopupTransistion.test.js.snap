// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PopupTransition should render 1`] = `
<DocumentFragment>
  .c0 {
  opacity: 1;
  -webkit-transform: translate(0,0%);
  -ms-transform: translate(0,0%);
  transform: translate(0,0%);
  -webkit-transition: all 0.6s -0.1s;
  transition: all 0.6s -0.1s;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.c0.popup-enter {
  -webkit-transform: translate(0,-100%);
  -ms-transform: translate(0,-100%);
  transform: translate(0,-100%);
  -webkit-transition: all 0.6s 0s;
  transition: all 0.6s 0s;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.c0.popup-exit {
  opacity: 0.9;
  -webkit-transform: translate(0,-130%);
  -ms-transform: translate(0,-130%);
  transform: translate(0,-130%);
  -webkit-transition: all 0.6s 0s;
  transition: all 0.6s 0s;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

<div
    class="c0"
  >
    Content
  </div>
</DocumentFragment>
`;
